### PR TITLE
[mesheryctl] Feature: `system reset --context/-c` to temporarily change context

### LINKED ISSUE #2285: [mesheryctl] support -c/--context in `system reset` command to temporarily override the current context
#### Current Behavior

Currently while running `mesheryctl system reset`, it automatically pulls the `docker-compose.yaml` file corresponding to the `current-context` mentioned in the `~/./meshery/config.yaml` file. Example-

```
contexts:
   local:
	endpoint: http://localhost:9081
	token: local-token
        platform: docker
        adapters: 
           - istio
           - linkerd
       channel: stable
       version: v0.4.20
   prod-us-east-1:
	endpoint: http://fiwaf-ssfdd.aws.useast1.amazon.com
	token: production-token
        platform: kubernetes
        channel: edge
        version: v0.4.27
current-context: prod-us-east-1
```

Now running `mesheryctl system reset` will automatically pull the latest `docker-compose.yaml` file from GitHub since the `channel` is `edge` based on the `current-context` which is `prod-us-east-1`.

#### Desired Behavior

Instead, the `system reset` command should also support overriding the current context with a `--context`, `-c` flag. 

So, for the above example, if we run `mesheryctl system reset -c local`, it should pull the `docker-compose.yaml` file from the `v0.4.20` release(since the `local` context uses the `stable` channel with version `v0.4.20`) instead pulling the latest file.

### PR DESCRIPTION
**Pull request by pottekkat**
Signed-off-by: navendu-pottekkat <navendupottekkat@gmail.com>

**Description**

This PR fixes #2285 | **This is a draft PR**

`mesheryctl system reset --context <context>` or `mesheryctl system reset -c <context>` will let users change the `current-context` temporarily while resetting the `meshery.yaml` file.

Example config.yaml file-

```
user@computer:~/.meshery$ cat config.yaml
contexts:
  local:
    endpoint: http://localhost:9081
    token:
      name: Default
      location: auth.json
    platform: docker
    adapters:
    - meshery-istio
    - meshery-linkerd
    - meshery-consul
    - meshery-octarine
    - meshery-nsm
    - meshery-kuma
    - meshery-cpx
    - meshery-osm
    - meshery-nginx-sm
    channel: stable
    version: v0.4.20
  prod:
    endpoint: http://localhost:9081
    token:
      name: Default
      location: auth.json
    platform: docker
    adapters:
    - meshery-istio
    - meshery-linkerd
    - meshery-consul
    - meshery-octarine
    - meshery-nsm
    - meshery-kuma
    - meshery-cpx
    - meshery-osm
    - meshery-nginx-sm
    channel: stable
    version: v0.4.22

current-context: local
tokens: {}
```

In the above file, the `current-context` is set to `local`. Now if the user wants to reset the `meshery.yaml` file based on the `prod` context, they can run `mesheryctl system reset -c prod` and it will temporarily set the `current-context` to `prod` **without** changing the `config.yaml` file.

```
user@computer:$ mesheryctl system reset --context prod
Meshery config file will be reset to system defaults. Are you sure you want to continue [y/n]? y
Meshery resetting...

Current Context: prod
Channel: stable
Version: v0.4.22

Fetching default docker-compose file at version: v0.4.22...

...Meshery config (/home/user/.meshery/meshery.yaml) now reset to default settings.
user@computer:$
```

And if the user tries to use an undefined `context`, it will throw an error-

```
user@computer:$ mesheryctl system reset --context test
Meshery config file will be reset to system defaults. Are you sure you want to continue [y/n]? y
Meshery resetting...

The specified context does not exist. The available contexts are:
prod
local
```

**Notes for Reviewers**

**This is a draft PR**

Please suggest how to handle invalid contexts.
Any other UX changes?

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Looking good so far

**Comment by leecalcote** [comment]
Still in draft, right?

**Comment by navendu-pottekkat** [comment]
@leecalcote Still in draft. It would be great if you can share how to handle if the users enter an invalid context. The current logic is like this-

> And if the user tries to use an undefined context, it will throw an error-

```
user@computer:$ mesheryctl system reset --context test
Meshery config file will be reset to system defaults. Are you sure you want to continue [y/n]? y
Meshery resetting...

The specified context does not exist. The available contexts are:
prod
local
```

Should I make any changes? 

Any other UX changes to make?

**Comment by navendu-pottekkat** [comment]
Now that I see it, maybe we should not log `Meshery resetting...` if the context is invalid

**Comment by leecalcote** [comment]
> Now that I see it, maybe we should not log `Meshery resetting...` if the context is invalid

You’re right. We should not log this. In addition, “resetting” is a misspelling.

**Comment by leecalcote** [comment]
> @leecalcote Still in draft. It would be great if you can share how to handle if the users enter an invalid context. The current logic is like this-
> 
> > And if the user tries to use an undefined context, it will throw an error-
> 
> ```
> user@computer:$ mesheryctl system reset --context test
> Meshery config file will be reset to system defaults. Are you sure you want to continue [y/n]? y
> Meshery resetting...
> 
> The specified context does not exist. The available contexts are:
> prod
> local
> ```
> 
> Should I make any changes?
> 
> Any other UX changes to make?

This looks pretty good to me. Phrasing in the last two sentences can be tighten up a little bit, like so:

```
“test” context does not exist. Available context(s):
local
prod
```

File: mesheryctl/internal/cli/root/system/reset.go [hunk 1]
```diff
@@ -53,11 +56,27 @@ func resetMesheryConfig() error {
 		}
 
 		currentContext := mctlCfg.CurrentContext
+
+		// if the user has mentioned a temporary context in the -c flag, change the context and proceed to reset
+		if tempContext != "" {
+			if _, val := mctlCfg.Contexts[tempContext]; val {
+				currentContext = tempContext
+			} else {
+				// if the user specifies a context that is not in the config.yaml file, throw an error and show the available contexts
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/reset.go:65]: Great comments here.

**Comment by navendu-pottekkat** [comment]
> This looks pretty good to me. Phrasing in the last two sentences can be tighten up a little bit, like so:
> 
> ```
> “test” context does not exist. Available context(s):
> local
> prod
> ```

@leecalcote This is perfect.

**Comment by navendu-pottekkat** [comment]
> In addition, “resetting” is a misspelling.

It is "resetting" right? reset--> resetting

**Comment by leecalcote** [comment]
@navendu-pottekkat, here is a test case to look into. Contents of `.meshery/config.yaml`:

```
contexts:
  local:
    endpoint: localhost:9081
    token:
      name: Default
      location: /Users/lee/.meshery/auth.json
    platform: ""
    adapters:
    - meshery-istio
    - meshery-linkerd
    - meshery-consul
    - meshery-octarine
    - meshery-nsm
    - meshery-kuma
    - meshery-cpx
    - meshery-osm
    - meshery-nginx-sm
    channel: edge
    version: bob
```

Command and output:

```
mesheryctl system reset
Meshery config file will be reset to system defaults. Are you sure you want to continue [y/n]? y
Meshery resetting...

Current Context: local
Channel: edge
Version: bob

Fetching default docker-compose file at version: bob...

...Meshery config (/Users/lee/.meshery/meshery.yaml) now reset to default settings.
```

**Comment by navendu-pottekkat** [comment]
@leecalcote 

Made the changes. Ready to merge.

```
user@computer:$ ./mesheryctl system reset -c new
Meshery config file will be reset to system defaults. Are you sure you want to continue [y/n]? y

"new" context does not exist. The available contexts are:
local
prod
test
user@computer:$
```

**Comment by leecalcote** [comment]
@navendu-pottekkat, there's a small lint check to have a look at - https://github.com/layer5io/meshery/pull/2288/checks?check_run_id=1757733079

**Comment by leecalcote** [comment]
I re-ran the CI checks, wondering if the lint check failure was a GitHub runner issue given that the error code isn't very description and doesn't point out a specific line in the code clearly. After a re-run, the lint issue still persists.

**Comment by navendu-pottekkat** [comment]
@leecalcote There are no lint errors on the files I worked on. But on VS Code I can see a lint error on another file. Maybe that is causing it.

**Comment by navendu-pottekkat** [comment]
@leecalcote Sorry! My bad. It was not showing error locally so I missed a blank line! No it is done!

**Pull request merged**